### PR TITLE
feat: accessibility and i18n: adds keyboard focus, canvas direction support

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
Adds mouse vs keyboard detection and BiDi hooks, updates ConfigProvider to consume useFocusVisible hook to enable accessible styling.

### CODE REVIEW

File: src/hooks/navigator.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+interface Navigator {
```
Comment (ychhabra-eightfold) [inline src/hooks/navigator.d.ts:1]: can we remove this ? if not move this to a more appropriate place

File: src/components/Button/button.module.scss [hunk 2]
```diff
@@ -536,6 +536,27 @@
     }
 }
 
+:global(.focus-visible) {
+    .button {
+        &:focus-visible {
+            outline-width: 4px;
```
Comment (ychhabra-eightfold) [inline src/components/Button/button.module.scss:542]: nit: dont hardcode px values

File: src/components/ConfigProvider/ConfigProvider.tsx [hunk 3]
```diff
@@ -11,6 +18,8 @@ const DEFAULT_FONT_SIZE = 16;
 
 const ConfigProvider: FC<ConfigProviderProps> = ({
     children,
+    focusVisible = false,
+    focusVisibleElement = document.documentElement,
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.tsx:22]: can focus visiblie have its own interface like themeoptions, will make this a bit cleaner

File: src/components/ConfigProvider/ConfigProvider.tsx [hunk 4]
```diff
@@ -24,6 +33,9 @@ const ConfigProvider: FC<ConfigProviderProps> = ({
         variableName: '--font-size',
     });
 
+    const isFocusVisible = useFocusVisible();
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.tsx:36]: nit: type

File: src/components/ConfigProvider/ConfigProvider.tsx [hunk 5]
```diff
@@ -24,6 +33,9 @@ const ConfigProvider: FC<ConfigProviderProps> = ({
...
+    const focusVisibleClassName: string = 'focus-visible';
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.tsx:37]: nit: Make this a constant at the top

File: src/components/ConfigProvider/ConfigProvider.tsx [hunk 6]
```diff
@@ -36,6 +48,30 @@ const ConfigProvider: FC<ConfigProviderProps> = ({
...
+            }
+        }
+    }, [isFocusVisible]);
+
+    useEffect(() => {
+        if (focusVisible) {
+            handleFocusVisible();
+        }
+    }, [handleFocusVisible]);
+
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.tsx:74]: Can this be a hook itself and be used here

File: src/hooks/useCanvasDirection.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    const getDirection = useCallback(
+        (lang: string) => {
+            if (document?.documentElement?.dir === 'rtl') {
+                setDir(document.documentElement.dir);
+            } else {
+                // only check first two chars of string to ensure no false positives (e.g. es-AR)
+                if (
+                    lang &&
+                    (lang.substring(0, 2) === 'ar' ||
+                        lang.substring(0, 2) === 'he')
```
Comment (ychhabra-eightfold) [inline src/hooks/useCanvasDirection.ts:20]:
```suggestion
                   lang?.substring(0, 2) === 'ar' || lang?.substring(0, 2) === 'he'
```

File: src/hooks/useCanvasDirection.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                } else {
+                    setDir('ltr');
+                }
+            }
+        },
+        [dir]
+    );
+
+    useEffect((): void => {
+        const lang =
```
Comment (ychhabra-eightfold) [inline src/hooks/useCanvasDirection.ts:32]: We might need an ability to pass lang as a prop as well

File: src/hooks/useCanvasDirection.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
+import { useCallback, useEffect, useState } from 'react';
+
```
Comment (ychhabra-eightfold) [inline src/hooks/useCanvasDirection.ts:2]: can we create a type ?
`type dir = 'inherit' | 'ltr' | 'rtl'`

File: src/hooks/useCanvasDirection.ts (hunk 7 above)
Comment (ychhabra-eightfold) [reply src/hooks/useCanvasDirection.ts:20]: `setDir(lang?.substring(0, 2) === 'ar' || lang?.substring(0, 2) === 'he' ? 'rtl' : 'ltr)`

File: src/components/Label/Label.tsx [hunk 10]
```diff
@@ -3,18 +3,22 @@ import { ButtonSize, DefaultButton } from '../Button';
...
 export const Label: FC<LabelProps> = ({
     classNames,
+    colon = false,
     htmlFor,
     labelIconButtonProps,
     size = LabelSize.Medium,
     text,
     ...rest
 }) => {
+    const htmlDir = useCanvasDirection();
```
Comment (ychhabra-eightfold) [inline src/components/Label/Label.tsx:20]: nit: type

File: src/hooks/useFocusVisible.ts [hunk 11]
```diff
@@ -0,0 +1,125 @@
+import { useEffect, useState } from 'react';
+
+// A TypeScript + React Hooks version of: https://github.com/WICG/focus-visible
+
+export const useFocusVisible = (): boolean => {
+    const [hadKeyboardEvent, setHadKeyboardEvent] = useState(false);
```
Comment (ychhabra-eightfold) [inline src/hooks/useFocusVisible.ts:6]:
```suggestion
    const [hadKeyboardEvent, setHadKeyboardEvent] = useState<bool>(false);
```

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/hooks/navigator.d.ts (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/hooks/navigator.d.ts:1]: Will move. This prevents a raised exception in IE.

File: src/components/ConfigProvider/ConfigProvider.tsx (hunk 6 above)
Comment (dkilgore-eightfold) [reply src/components/ConfigProvider/ConfigProvider.tsx]: This logic is written for the ConfigProvider. Shouldn't it remain here for testability? It relies on the props to function.

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**